### PR TITLE
MM-12996/MM-13229 - Updating modals on mobile and fixing plugin button on mobile

### PR DESCRIPTION
**Pull request by asaadmahmood**
#### Summary
MM-12996 - Updating modals on mobile

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12996
https://mattermost.atlassian.net/browse/MM-13229

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Ran `make check-style` to check for style errors (required for all pull requests)
- [ ] Ran `make test` to ensure unit and component tests passed
- [ ] Added or updated unit tests (required for all new features)
- [ ] Needs to be implemented in mobile (link to PR or User Story)
- [ ] Has server changes (please link)
- [ ] Has redux changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

File: plugins/mobile_channel_header_plug/mobile_channel_header_plug.jsx [hunk 1]
```diff
@@ -28,14 +28,16 @@ export default class MobileChannelHeaderPlug extends React.PureComponent {
...
         return (
-            <button
-                className='navbar-toggle navbar-right__icon'
-                onClick={() => this.fireAction(plug)}
-            >
-                <span className='icon navbar-plugin-button'>
-                    {plug.icon}
-                </span>
-            </button>
+            <li className='flex-parent--center'>
```
Comment (sudheerDev) [inline plugins/mobile_channel_header_plug/mobile_channel_header_plug.jsx]: nit: should we be using li when we don't have a ul/ol?

Comment (asaadmahmood) [reply plugins/mobile_channel_header_plug/mobile_channel_header_plug.jsx]: The <ul> is there on the parent for hte whole dropdown menu.

**Comment by hmhealey** [comment]
I've marked this as DO NOT MERGE since the changes to the MobileChannelHeaderPlug make the tests not test anything and because I'm still seeing this issue

**Comment by hmhealey** [comment]
Sorry, I was testing the wrong thing. The linked ticket has been fixed, although there's also some fixes for a different bug that need a ticket created

**Comment by asaadmahmood** [comment]
Talked to @hmhealey, we're going with the same PR for both those tickets since it's already looked at, and we're quite near code complete.
Though, I know we have strict guidelines to keep a PR per ticket.

**Pull request merged**